web/js/util/util.js

```diff
@@ -293,6 +294,33 @@ export default (function(self) {
     return date;
   };
 
+  self.coverageDateFormatter = function(dateType, date, period) {
+    let dateString;
+    date = this.parseDate(date);
+
+    switch (period) {
+      case 'subdaily':
+        dateString = moment(date).format('DD MMMM YYYY HH:mm') + 'Z';
+        break;
+
+      case 'yearly':
+        if (dateType === 'END-DATE') date.setFullYear(date.getFullYear() - 1);
```

Comment:
This looks interesting. Why do we have to subtract a year or a month from end dates for those periods?

Reply:
I had trouble wrapping my head around it - but I carried this over from the previous code base because on those set periods, the end-date for that layer is off by a single day. For example, the BLACK MARBLE 2012-2016 layer has a end date of January 1st, 2017; however, the data coverage technically stops then on December 31st, 2016. Since it has a period of a year, the formatting would look like 2012-2017 when it should be 2012-2016; hence the subtraction. 

This is the same for months. Using the same dates above, we do NOT want January 2012 - January 2017; rather, we would like it to be January 2012 - December 2016.

Reply:
Ah that's right, I remember the discussion on slack the other day about this.  Well, if we're already doing this and it works, that works for me 👍